website/pages/docs/secrets/key-management/index.mdx

```diff
@@ -0,0 +1,189 @@
+---
+layout: docs
+page_title: Key Management - Secrets Engines
+sidebar_title: Key Management <sup>ENTERPRISE</sup>
+description: >-
+  The Key Management secrets engine provides a consistent workflow for distribution and lifecycle
+  management of cryptographic keys in various key management service (KMS) providers.
+---
+
+# Key Management Secrets Engine
+
+-> **Note**: This secrets engine requires [Vault
+Enterprise](https://www.hashicorp.com/products/vault/) with the Advanced Data
+Protection Module.
+
+The Key Management secrets engine provides a consistent workflow for distribution and lifecycle
+management of cryptographic keys in various key management service (KMS) providers. It allows
+organizations to maintain centralized control of their keys from Vault while still taking advantage
+of cryptographic capabilities native to the KMS providers.
+
+The secrets engine generates and owns original copies of key material. When an operator decides
+to distribute and manage the lifecycle of a key in one of the [supported KMS providers](#kms-providers),
+a copy of the key material is distributed. This provides additional durability and disaster
+recovery means for the complete lifecycle of the key in the KMS provider.
+
+Key material will always be securely transferred in accordance with the
+[key import specification](#kms-providers) of the supported KMS providers.
+
+## Setup
+
+Most secrets engines must be configured in advance before they can perform their
+functions. These steps are usually completed by an operator or configuration
+management tool.
+
+1. Enable the Key Management secrets engine:
+
+    ```text
+    $ vault secrets enable keymgmt
+    Success! Enabled the keymgmt secrets engine at: keymgmt/
+    ```
+
+    By default, the secrets engine will mount at the name of the engine. To enable
+    the secrets engine at a different path, use the `-path` argument.
+
+## Usage
+
+After the secrets engine is mounted and a user/machine has a Vault token with
+the proper permission, it can use this secrets engine to generate, distribute, and
+manage the lifecycle of cryptographic keys in [supported KMS providers](#kms-providers).
+
+1. Create a named cryptographic key of a specified type:
+
+    ```text
+    $ vault write -f keymgmt/key/example-key type="rsa-2048"
+    Success! Data written to: keymgmt/key/example-key
+    ```
+
+    Keys created by the secrets engine are considered general-purpose until
+    they're distributed to a KMS provider.
+
+1. Configure a KMS provider:
+
+    ```text
+    $ vault write keymgmt/kms/example-kms \
+        provider="azurekeyvault" \
+        key_collection="keyvault-name" \
+        credentials=client_id="" \
```

Comment:
Should we provide dummy values for credentials?

Reply:
Added fake credentials to the example in 9e2e8a5. Let me know if you'd prefer these be something more like `client_id="example-client-id"` instead.

Reply:
I think what you have is fine (similar to [aws sample values](https://www.vaultproject.io/api/auth/aws#sample-payload) in the docs)! It's mostly to indicate that these should be non-empty values.